firedrake/interpolation.py

```diff
@@ -215,8 +250,58 @@ def __init__(
         self._allow_missing_dofs = allow_missing_dofs
         self.callable = None
 
+    @PETSc.Log.EventDecorator()
+    def interpolate(self, *function, output=None, transpose=False, default_missing_val=None):
+        """This performs interpolation by assembling `Interp`, which in turn calls
+        `Interpolator._interpolate`. Having this structure ensures consistency between
+        `Interpolator` and `Interp`.
+
+        This mechanism handles annotation since performing interpolation will drop an
+        Assemble block on the tape.
```

Comment:
This is implementation detail, rather than a docstring. This docstring should be identical to the existing docstring for `Interpolator.interpolate`, since the public interface has not changed. Details like this should be in a comment.